presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -140,14 +140,14 @@
         this.subqueryPlanner = new SubqueryPlanner(analysis, variableAllocator, idAllocator, lambdaDeclarationToVariableMap, metadata, session);
     }
 
-    public RelationPlan plan(Query query)
```

Comment:
We can store context in QueryPlanner and SubqueryPlanner as class variables, instead of just passing around

Reply:
Good suggestion!
Did this for `QueryPlanner `. For `SubqueryPlanner` it's a bit more challenging as the RelationPlanner has `SubqueryPlanner` as its class member, and RelationPlanner is a AST visitor which takes the context as function query parameters, which is not available when constructing RelationPlanner.